x-pack/plugins/cases/public/components/all_cases/utils.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { parse } from 'query-string';
+
+import type { ParsedUrlQueryParams, UrlQueryParams } from '../../../common/ui/types';
+
+export const parseUrlQueryParams = (search: string): UrlQueryParams => {
+  const parsedUrlParams: ParsedUrlQueryParams = parse(search);
+
+  const urlParams: UrlQueryParams = {
+    ...(parsedUrlParams.sortField && { sortField: parsedUrlParams.sortField }),
+    ...(parsedUrlParams.sortOrder && { sortOrder: parsedUrlParams.sortOrder }),
+  };
+
+  const intPage = parsedUrlParams.page && parseInt(parsedUrlParams.page, 10);
```

Comment:
Maybe we can use `Number(parsedUrlParams.page)`. Also, we should check if the resulting number is `NaN` and act accordingly. `parsedUrlParams.page` is controlled by the user and can be anything.

Reply:
> Also, we should check if the resulting number is NaN and act accordingly. parsedUrlParams.page is controlled by the user and can be anything.

The ifs below were to take that into account, if NaN we ignore the URL page value.